pkg/scheduler/queue/leafqueue.go

```diff
@@ -78,7 +75,7 @@ func (q *LeafQueue) Dequeue() Request {
 	}
 
 	// only if there are no items queued in the local queue, dequeue from sub-queues
-	maxIter := q.mapping.Len()
+	maxIter := len(q.mapping.keys)
```

Comment:
Wasn't the old `q.mapping.Len()` correct since it subtracted empty keys?

Reply:
Imagine `keys=[x, x, "foo", "bar", x]` where x marks the empty slot. When you are at index=3 and want to get the next items.
If you use `Len()`, then you would only iterate to index=0, which is empty and therefore would not return a sub queue.